internal/api/middleware.go

```diff
@@ -84,3 +89,76 @@ func loggingMiddleware(ctx *middlewareContext) middleware {
 		})
 	}
 }
+
+// authMiddleware intercepts requests that need authorization to check if the user's token exists and is
+// permitted to use the endpoint
+func authMiddleware(ctx *middlewareContext) middleware {
+	AdminOnlyPaths := []struct{ method, path string }{
+		{"POST", `accounts`},
+		{"GET", `accounts`},
+		{"GET", `accounts\/\d+$`},
+		{"DELETE", `accounts\/\d+$`},
+		{"POST", `accounts\/\d+\/change_password$`},
+	}
+
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if !strings.HasPrefix(r.URL.Path, "/api/v1/") {
+				next.ServeHTTP(w, r)
+				return
+			}
+			if r.Method == "POST" && strings.HasSuffix(r.URL.Path, "accounts") && !ctx.firstAccountIssued {
```

Comment:
Nitpicking but does the `r.Method == "POST"` part have any value here?
It might be a nicer experience to get an empty list of user when sending a GET to /accounts when no users are created yet.

Reply:
just to be clear are you saying users who haven't logged in yet should be able to get the list of accounts if no account has been created yet?

Reply:
Yes, only if no accounts have been created
But it doesn't make any difference, as the result will be no accounts in both cases
I only think it could be a nicer experience
But as I think of it a bit more I realise it doesn't bring any added value, so ignore it if you want :)

Reply:
I think it would be a bit weird to have that send an empty response instead of a 403 for the entire `/api/v1`. The POST request is needed but this one is not so it may give the wrong impression.